File: src/uiTest/java/org/assertj/swing/matcher/IdeaTableViewMatcher.java
package org.assertj.swing.matcher;

import com.intellij.ui.table.TableView;
import org.assertj.swing.core.GenericTypeMatcher;

public class IdeaTableViewMatcher extends GenericTypeMatcher<TableView> {
    public IdeaTableViewMatcher() {
        super(TableView.class);
    }

    @Override
    protected boolean isMatching(TableView component) {
        return true;
    }

    @Override
    public String toString() {
        String format = "%s[]";
        return String.format(format, getClass().getName());
    }
}

File: src/uiTest/kotlin/org/assertj/swing/matcher/IdeaTableViewMatcher.kt
package org.assertj.swing.matcher

import com.intellij.ui.table.TableView
import org.assertj.swing.core.GenericTypeMatcher

class IdeaTableViewMatcher
    : GenericTypeMatcher<TableView<*>>(TableView::class.java) {

    override fun isMatching(component: TableView<*>): Boolean {
        return true
    }

    override fun toString(): String {
        val format = "%s[]"
        return String.format(format, javaClass.name)
    }
}
